### PR TITLE
Update linux-common role to support Infiniband netdevs

### PR DESCRIPTION
**Pull request by tomclark0**
Introduces a linux-common role, used by all linux based images for changes from default source images.

This role is initially used to provide InfiniBand support using inbox drivers from the relevant distro.
It may later be serve other uses cases where additional packages are required as a baseline, e.g for diagnostic uses.

The default cloud-init network config renderer for Ubuntu (netplan) does not support InfiniBand type devices. To mitigate this, the renderer priority list is changed to match EL9 distros.

The required package for InfiniBand inbox drivers in Ubuntu is kernel-module-extra-$(uname -r). This is a per-kernel package, i.e any kernel update requires the explicit installation of the new version. 
To support kernel updates within an image, the linux-generic metapackage is used to provide tracking across kernel versions.

EL9 based distros require only additional packages and module loading.

### CODE REVIEW

File: ansible/linux-webconsole.yml [hunk 1]
```diff
@@ -3,4 +3,5 @@
 - hosts: all
   become: yes
   roles:
+    - linux-common
```
Comment (JohnGarbutt) [inline ansible/linux-webconsole.yml:6]: This is looking good to me, but clearly we need a rebase and we need to include the K8s builds.
After the rebase I think the K8s builds are already using linux-common, so that might just work 🤞

**Comment by tomclark0** [comment]
Changing the renderer without changing the activator led to the network interface bouncing after network-online.target is reached.
For services that launch after network-online.target this can be problematic.

File: ansible/roles/linux-common/tasks/main.yml [hunk 2]
```diff
@@ -26,3 +25,52 @@
...
     name: linux-volumes-by-tag
+
+# Use linux-generic to track linux-modules-extra
+- name: Install InfiniBand support (deb based)
+  ansible.builtin.apt:
+    name: linux-generic
+    state: latest
+    update_cache: yes
+  when: ansible_os_family == "Debian"
+  become: true
```
Comment (mkjpryor) [inline ansible/roles/linux-common/tasks/main.yml:36]: All these `become`s are not necessary, because the playbooks specify `become: yes` so all tasks are executed with root privileges unless otherwise configured on the task.

File: ansible/roles/linux-common/tasks/main.yml [hunk 3]
```diff
@@ -26,3 +25,52 @@
...
+- name: Switch cloud-init to alternative network activator
+  lineinfile:
+    path: /etc/cloud/cloud.cfg
+    regexp: "^    activators:.*"
+    line: "    activators: ['networkd', 'eni', 'network-manager']"
+    state: present
+  when: ansible_os_family == "Debian"
+  become: true
+
+
```
Comment (mkjpryor) [inline ansible/roles/linux-common/tasks/main.yml:71]: Nit - unnecessary whitespace.

**Review by mkjpryor — CHANGES_REQUESTED** [review]
We currently only support Ubuntu-based images with this code, so having all the `os_family` constraints (and the Red Hat install task) is currently unnecessary. I don't think this is likely to change in the near future.

**Review by mkjpryor — APPROVED** [review]
LGTM
